airflow/contrib/kubernetes/pod_generator.py

```diff
@@ -0,0 +1,290 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from airflow.contrib.kubernetes.pod import Pod, Resources
+
+
+class PodGenerator:
+    """Contains Kubernetes Airflow Worker configuration logic"""
+
+    def __init__(self, kube_config=None):
+        self.kube_config = kube_config
+        self.env_vars = {}
+        self.volumes = []
+        self.volume_mounts = []
+        self.init_containers = []
+        self.secrets = []
+
+    def set_environment(self, env):
```

Comment:
Personal gripe time (feel free to ignore):
getters/setters in python that just wrap a property are usually bad practice. is there a good reason we need these (I don't see where this setter is used).